src/scheduler/create-or-update-job.controller.spec.js

```diff
@@ -0,0 +1,167 @@
+        scope = $rootScope.$new();
+        scope.action = 'edit';
+        controller = $controller('CreateOrUpdateJobController', {
+            $scope: scope,
+            JobsService: _JobsService_,
+            $stateParams: { jobName: null },
+            ModalFactory: _ModalFactory_,
+            LoadingModal: _LoadingModal_,
+            MotechAlert: MotechAlertSpy,
+        });
+        JobsService = _JobsService_;
+        ModalFactory = _ModalFactory_;
+        LoadingModal = _LoadingModal_;
+        i18nService = _i18nService_;
+    }));
+
+    it('should set scope.jobTypes', function() {
+        expect(scope.jobTypes).toEqual([
+            { displayName: "Cron", name: "CRON" },
+            { displayName: "Repeating", name: "REPEATING" },
+            { displayName: "Repeating Period", name: "REPEATING_PERIOD"},
+            { displayName: "Run Once", name: "RUN_ONCE" },
+            { displayName: "Day of Week", name: "DAY_OF_WEEK" }
+        ]);
+    });
+
+    it('should set scope.days', function() {
+        expect(scope.days).toEqual([
+            { label: "Monday", value: "0" },
+            { label: "Tuesday", value: "1" },
+            { label: "Wednesday", value: "2" },
+            { label: "Thursday", value: "3" },
+            { label: "Friday", value: "4" },
+            { label: "Saturday", value: "5" },
+            { label: "Sunday", value: "6" }
+        ]);
+    });
+
+    describe('resetMap method', function () {
+        it('should show confirm', function () {
+            spyOn(ModalFactory, 'showConfirm');
+            scope.motechEventParameters = { test: 'test' };
+            scope.resetMap();
+
+            expect(ModalFactory.showConfirm).toHaveBeenCalled();
+        });
+    });
+
+    describe('getMinDate method', function () {
+        it('should return formatted date', function () {
+            expect(scope.getMinDate("RUN_ONCE")).not.toBeNull();
+            expect(scope.getMinDate("RUN_ONCE")).toBeTruthy();
+        });
+
+        it('should return null', function () {
+            expect(scope.getMinDate()).toBeNull();
+        });
+    });
+
+    describe('create or update job method', function () {
+        it('should execute MotechAlert with with good parameters if scope.dates.startDate greater than scope.dates.endDate', function () {
+            var currentDate = new Date();
+            scope.dates.startDate = scope.parseDateToString(currentDate);
+            scope.dates.endDate = scope.parseDateToString(currentDate.addDays(1));
+
+            scope.createOrUpdateJob();
+
+            expect(MotechAlertSpy).not.toHaveBeenCalledWith(
+                i18nService.getMessage("scheduler.error.endDateBeforeStartDate",
+                                       [scope.dates.startDate, scope.dates.endDate]), "scheduler.error");
+        });
+
+        it('should not execute MotechAlert with with good parameters if scope.dates.endDate greater than scope.dates.startDate', function () {
```

Comment:
rename the test to: 'should not execute MotechAlert with good parameters if scope.dates.startDate is after scope.dates.endDate'